doc/conf.py

```diff
@@ -308,29 +308,93 @@ def _check_dependencies():
 latex_logo = None
 
 latex_elements = {}
+
+latex_engine = 'xelatex'  # many Unicode glyphs in our docs...
+# By default Sphinx with latex_engine = 'xelatex' uses polyglossia
+# If this key is changed, clean the build/latex directory before next build
+latex_elements['babel'] = r'\usepackage{babel}'
+
 # Additional stuff for the LaTeX preamble.
+
+# one line per author name on title page
 latex_elements['preamble'] = r"""
-   % In the parameters section, place a newline after the Parameters
-   % header.  (This is stolen directly from Numpy's conf.py, since it
-   % affects Numpy-style docstrings).
-   \usepackage{expdlist}
-   \let\latexdescription=\description
-   \def\description{\latexdescription{}{} \breaklabel}
-
-   \usepackage{amsmath}
-   \usepackage{amsfonts}
-   \usepackage{amssymb}
-   \usepackage{txfonts}
-
-   % The enumitem package provides unlimited nesting of lists and
-   % enums.  Sphinx may use this in the future, in which case this can
-   % be removed.  See
-   % https://bitbucket.org/birkenfeld/sphinx/issue/777/latex-output-too-deeply-nested
-   \usepackage{enumitem}
-   \setlistdepth{2048}
+  \DeclareRobustCommand{\and}{%
+    \end{tabular}\kern-\tabcolsep\\\begin{tabular}[t]{c}%
+  }%
 """
+
+# Usage of expdlist package (as originally copied over from Numpy's confpy)
+# is removed because it breaks the PDF build with recent LaTeX:
+#    ! Extra }, or forgotten \endgroup.
+#      \@endpbox ...\@arstrutbox \color@endgroup \egroup
+# The https://github.com/scipy/scipy/pull/8657 code would actually fix this
+#
+# latex_elements['preamble'] = r"""
+#    % In the parameters section, place a newline after the Parameters
+#    % header.  (This is stolen directly from Numpy's conf.py, since it
+#    % affects Numpy-style docstrings).
+#    \usepackage{expdlist}
+#    \let\latexdescription=\description
+#    \def\description{\latexdescription{}{} \breaklabel}
+# """
+
 latex_elements['pointsize'] = '11pt'
 
+# Fix fancyhdr complaining about \headheight too small
+latex_elements['passoptionstopackages'] = r"""
+    \PassOptionsToPackage{headheight=14pt}{geometry}
+"""
+
+# Sphinx 1.5 provides this to avoid "too deeply nested" LaTeX error
+# and usage of "enumitem" LaTeX package is unneeded.
+# Value can be increased but do not set it to something such as 2048
+# which needlessly would trigger creation of thousands of TeX macros
+latex_elements['maxlistdepth'] = '10'
+
+# Use OpenType font with wide-enough glyph coverage
+# Sphinx 2.0 adopts GNU FreeFont by default, but it does not have all
+# the Unicode codepoints needed for the section about Mathtext
+# "Writing mathematical expressions"
+latex_elements['fontpkg'] = r"""
+\setmainfont{XITS}[
```

Comment:
On one hand, I would prefer to stick to dejavu to match the default font of our figurse, on the other hand the website does not do that and it may be a tad too cute.

Making the build cleaner seems like a greater good, than the font consistency.

Reply:
All of this is related to the documentation of "Mathtext". It contains extensive tables of math symbols, matching Unicode code points to LaTeX-like math markup. For the PDF it appears thus needed that the text font contains glyphs for all those Unicode code points. DejaVu was lacking 134 of them (``grep "There is no" Matplotlib.log | sort -u``).

There is still some piece of the documentation which will renders non satisfactorily because it is supposed to illustrate various fonts, but the PDF uses only one! (it is possible probably, but complicated and will need ``raw:: latex``  directives, to fix that).